consensus/wal.go

```diff
@@ -12,18 +17,29 @@ import (
 //--------------------------------------------------------
 // types and functions for savings consensus messages
 
+var (
+	walSeparator = []byte{55, 127, 6, 130} // 0x377f0682 - magic number
```

Comment:
what happens when this sequence shows up in a block ?

Reply:
If we're using our old group#Search method or `gr.ReadLine`, we would definitely get bad results. Imagine for some reason we failed to write a separator "\n" or it's lost due to FS errors, then if we use `gr.ReadLine` we'll get 2 messages, but only parse one.

```
{"time":"2017-10-24T08:13:04.564Z","msg":{"height":6,"round":0,"step":"RoundStepNewHeight"}}{"time":"2017-10-24T08:13:04.564Z","msg":{"height":7,"round":0,"step":"RoundStepNewHeight"}}\n
```

new WAL#SearchForHeight method sequentially reads the WAL file and panics if no separator is found.

> what happens when this sequence shows up in a block ?

answering your original question, nothing bad will happen because as I explained above we're no longer splitting the log (using `strings.Split` or `bytes.Split`; `bytes.Split` is only used in tests), but rather read it byte by byte (4bytes CRC -> 4 bytes length -> (length) data -> separator -> 4bytes CRC -> 4 bytes length ...)

Reply:
Yep thanks